extensions/blocks/recurring-payments/edit.jsx

```diff
@@ -459,4 +504,20 @@ class MembershipsButtonEdit extends Component {
 	};
 }
 
-export default withNotices( MembershipsButtonEdit );
+export default compose( [
+	withSelect( select => ( { postId: select( 'core/editor' ).getCurrentPostId() } ) ),
+	withDispatch( dispatch => ( {
+		autosaveAndNavigateToConnection: async event => {
+			const { href, target } = event.target;
+
+			// Special handling if we're opening in _this_ window. Otherwise, just let the navigation happen.
+			if ( href && target !== '_blank' ) {
+				event.preventDefault();
+				await dispatch( 'core/editor' ).autosave();
+				// Using window.top to escape from the editor iframe on WordPress.com
+				window.top.location.href = href;
```

Comment:
Do we need to guard here for potentially undefined window? 
cc @ockham

Reply:
We have a [redirect like this](https://github.com/Automattic/jetpack/blob/a9959f247ae98b07aeb5a3c1814477c30a0744ec/extensions/shared/components/block-nudge/index.jsx#L45) in the block nudge already. I'm not sure it's dangerous to have `window` unguarded in cases like this; while we do use the block nudge for server side rendering, it's only the plain 'dumb' component rather than the `withDispatch()` case, so we're not actually ever using it in a context where `window` doesn't exist -- and I think we'd notice pretty much immediately if we did, since it'll totally break the build. (The same argument applies to this occurrence.) 

My take would thus be YAGNI -- let's only add it once we need it, e.g. if the build fails.

Reply:
Ok, makes sense I guess. I am not familiar with the server side rendering stuff - just curious as you mentioned it previously. So YAGNI - if the build fails, then fix? Not sure that was the case previously tbh (I don't remember builds failing, but it was a fairly different case + may be wrong/missing something). I'm clueless on the ssr stuff. :-)

Reply:
It's a good question, guarding browser globals is cheap and saves pain down the road. The problem here is that none of this block makes sense outside the browser. We can't `event.preventDefault` if we can't handle the redirect. That leads to being unable to save the content before navigation, and at that point we might as well just error.

Reply:
What Jon said -- it's just a `noop` outside the browser (we do however have an `href` in the 'dumb' component as the fallback for the SSR case, so a `noop` should be okay -- but we essentially do that, at least in the Upgrade Nudge case, by _using_ the dumb rather than the smart component in SSR).